core/commands/publish.go

```diff
@@ -71,21 +72,32 @@ Publish a <ref> to another public key:
 			return
 		}
 
-		// name := ""
-		ref := ""
+		var pstr string
 
 		switch len(args) {
 		case 2:
 			// name = args[0]
-			ref = args[1]
+			pstr = args[1]
 			res.SetError(errors.New("keychains not yet implemented"), cmds.ErrNormal)
 		case 1:
 			// name = n.Identity.ID.String()
-			ref = args[0]
+			pstr = args[0]
+		}
+
+		node, err := n.Resolver.ResolvePath(path.FromString(pstr))
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		hash, err := node.Multihash()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
 		}
 
 		// TODO n.Keychain.Get(name).PrivKey
-		output, err := publish(n, n.PrivateKey, ref)
+		output, err := publish(n, n.PrivateKey, hash.B58String())
```

Comment:
can use:

```go
key, err := node.Key()
if err != nil {
    res.SetError(err, cmds.ErrNormal)
    return
}

output, err := publish(n, n.PrivateKey, key.Pretty())
```

So that if we change our "key pretty-printing" it will change here too.